### PR TITLE
feat: first implementation of data-to-array

### PR DESCRIPTION
**Pull request by thomas9911**

### CODE REVIEW

File: functions/parse-to-array/1.0/index.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const parseToArray = async ({ data, dataType }) => {
+  const parsedData = await parseData({ data, format: dataType });
+
```
Comment (thomas9911) [inline functions/parse-to-array/1.0/index.js:3]: should we do a check here to validate that the parsedData is an array?

File: functions/parse-to-array/1.0/function.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    },
+    {
+      "name": "data",
+      "label": "Data",
+      "meta": {
+        "type": "Value",
+        "validations": {
+          "required": true
+        },
+        "allowedKinds": ["STRING"]
```
Comment (thomas9911) [inline functions/parse-to-array/1.0/function.json:40]: We should check if this also allows that files can be used as input

File: functions/parse-to-array/1.0/function.json [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  "description": "Parses the data as an array of the given Schema Model",
+  "label": "Parse Data To Array",
+  "category": "Misc",
+  "icon": {
+    "name": "DataLineIcon",
+    "color": "Purple"
+  },
+  "options": [
+    {
+      "info": "Enter what protocol the receiving end needs to use. You can choose between HTTP and HTTPS, which one you should use depends on your host.",
```
Comment (thomas9911) [inline functions/parse-to-array/1.0/function.json:11]: recheck this blueprint if the info is correct

File: functions/parse-to-array/1.0/function.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
+{
+  "description": "Parses the data as an array of the given Schema Model",
+  "label": "Parse Data To Array",
```
Comment (emiel-betty) [inline functions/parse-to-array/1.0/function.json:3]: Is this the correct label? Lets discuss with PO.

File: functions/parse-to-array/1.0/function.json [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        "validations": {
+          "required": true
+        },
+        "output": {
+          "type": "Array",
+          "dataType": "SCHEMA",
+          "schemaModel": "schemaModel"
+        }
+      },
+      "info": "Choose a name for the variable under which the unique value which is grouped on will be available.",
```
Comment (emiel-betty) [inline functions/parse-to-array/1.0/function.json:67]: Please check the info text for each option. This is not correct.

File: functions/parse-to-array/1.0/index.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+const isArray = (data) => typeof data === 'object' && 'length' in data;
+
+const parseToArray = async ({ data, dataType }) => {
+  const parsedData = await parseData({ data, format: dataType });
+
+  if (isArray(parsedData)) {
```
Comment (emiel-betty) [inline functions/parse-to-array/1.0/index.js:6]: You can use `Array.isArray ( arg )`

File: functions/parse-to-array/1.0/function.json [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+          {
+            "label": "CSV",
+            "value": "CSV"
+          }
+        ]
+      },
+      "name": "dataType"
+    },
+    {
+      "info": "The actual data, can be text or a file",
```
Comment (archan937) [inline functions/parse-to-array/1.0/function.json:33]: "can be text or a URL" right?

**Review by archan937 — APPROVED** [review]

**Pull request merged**